### ISSUE TITLE
Can not cross compile on Max OS X (Mavericks)

### ISSUE THREAD

**Issue by raphaelcohn** [landley/toybox#17]
* `make menuconfig` works
* `make` fails because a `sed` expression isn't understood by Mac OS X (and hence BSD) `sed`
It may be possible to have a different expression. If you can give me a little insight into what the expression is trying to do, I can have a go at sorting this out.

I'm really keen to get a minimal toybox working on other POSIX platforms, because it provides an excellent baseline for a consistent, secure and sensible set of tools to enable cross compilation toolchains. Oh, and not using autocruft gets it an extra star...

More details:-
```bash
maclaptop:toybox raph$ make
scripts/make.sh
Generate headers from toys/*/*.c...
Make generated/config.h from .config.
generated/flags.h sed: 2: "s/" *"//g;/^#/d;t clear ...": undefined label 'clear;:clear;s/"/"/p;t;s/\( [AB] \).*/\1 " "/p'
sed: 1: "s/ A / /;t pair;h;s/\([ ...": undefined label 'pair;h;s/\([^ ]*\).*/\1 " "/;x;b single;:pair;h;n;:single;s/[^ ]* B //;H;g;s/\n/ /;p'
clang: error: unable to execute command: Broken pipe: 13
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix
clang: note: diagnostic msg: PLEASE submit a bug report to http://developer.apple.com/bugreporter/ and include the crash backtrace, preprocessed source, and associated run script.
clang: note: diagnostic msg: Error generating preprocessed source(s) - ignoring input from stdin.
clang: note: diagnostic msg: Error generating preprocessed source(s) - no preprocessable inputs.
sed: 2: "s/" *"//g;/^#/d;t clear ...": undefined label 'clear;:clear;s/"/"/p;t;s/\( [AB] \).*/\1 " "/p'
clang: error: unable to execute command: Broken pipe: 13
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix
clang: note: diagnostic msg: PLEASE submit a bug report to http://developer.apple.com/bugreporter/ and include the crash backtrace, preprocessed source, and associated run script.
clang: note: diagnostic msg: Error generating preprocessed source(s) - ignoring input from stdin.
clang: note: diagnostic msg: Error generating preprocessed source(s) - no preprocessable inputs.
Makefile:19: recipe for target 'toybox' failed
make: *** [toybox] Error 1
```

**Comment by landley** [comment]
On 12/15/2015 06:30 AM, Raphael Cohn wrote:
>   * |make menuconfig| works

Huh, when I tried it on freebsd and pcbsd, the kconfig stuff was the
blocker. (I have a todo item to rewrite the kconfig stuff from scratch,
but it's pretty far down the todo list because what I copied from
linux.2.6.12 years ago still works...)

>   * |make| fails because a |sed| expression isn't understood by Mac OS X
>     (and hence BSD) |sed| It may be possible to have a different
>     expression. If you can give me a little insight into what the
>     expression is trying to do, I can have a go at sorting this out.

Before I respond more, could you send this to the mailing list instead
of through github? I try to keep development discussions there.

(Or could I just cc: my reply to the mailing list?)

Thanks,

Rob

**Comment by landley** [comment]
On 01/07/2016 05:43 AM, Raphael Cohn wrote:
> Closed #17 <https://github.com/landley/toybox/issues/17>.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/landley/toybox/issues/17#event-508149436>.

Could you email me directly instead of through github? (Or to the toybox
list.) I tend to lose track of things coming in from github.

I'm interested in merging the cyanogenmod work upstream, but the odd
thing about toybox being under a public domain license
(http://spdx.org/licenses/0BSD.html) is that I can't take patches
without asking because they could be under any license (or none). It's
_implied_ that they're under the same license as the original project
unless stated otherwise, but I tend to hold to a higher standard than that.

(Technically GPLv2 or later projects could get GPLv3 patches they
couldn't pull back either. Those guys never police it. I did the busybox
lawsuits way back when and thus have... scars. Same way that In theory
work gave me a mac laptop I can test builds on. In practice it won't let
me run "git" unless I agree to a clickthrough contract first and I
object in principle, I installed a linux vm on it instead.)

I actually looked up cyanogenmod's git repo for toybox and there's a
bunch of patches fixing issues that I'll fix in a different way since
they weren't submitted to me. Should be in the next release.

Thanks,

Rob
